### DISCUSSION TITLE
Storing consumer offsets in a different place

### DISCUSSION THREAD

**Discussion by nanofish** [spring-projects/spring-kafka#2177] [Q&A]
Hi, Is there support for storing consumer offsets in a different place (eg a database?). Something like an interface that is called for offset retrieval on partition assignment and offset storage on acknowledge?

**Comment by garyrussell** [comment]
You can do whatever you want; just set the container ack mode to `MANUAL` and `AssignmentCommitOption` to `NEVER` so that the container will never commit any offsets.

You can then have your listener extend `AbstractConsumerSeekAware` where you will get assignment calls and can perform seeks.

Add the partition and offset as parameters to your listener method to get the information for the current record.

https://docs.spring.io/spring-kafka/docs/current/reference/html/#seek
